### ISSUE TITLE
from c-lightning:  Could not find a route

### ISSUE THREAD

**Issue by charltonh** [btcontract/lnwallet#18]
I have not been able to use BLW for months, as even though it has a direct channel to my c-lightning node, c-lightning cannot find it to route a simple channel refill payment to.

I've read somewhere that this is an eClair issue, about broadcasting node information.  When is it going to be fixed?  Will I ever be able to use BLW again or should I just uninstall it and take it off the list of recommended wallets?

**Comment by btcontract** [comment]
Are you trying to send a direct payment from cl to blw or routed payment from don't other place?

**Comment by fiatjaf** [comment]
I believe this is a c-lightning bug with channels in route hints. Maybe related to https://github.com/ElementsProject/lightning/issues/3613 or https://github.com/ElementsProject/lightning/issues/3607

**Comment by charltonh** [comment]
> Are you trying to send a direct payment from cl to blw or routed payment from some other place?

I'm trying to send a direct payment from cl to blw in this case.

Since there's a direct channel/connection to the node, it is a bit ridiculous that it cannot find a route.  So I disassembled the bolt11 and sure enough there's nothing in it to indicate to cl where it came from.   If cl cannot pay blw, isn't that a major issue?  The world needs lightning right now, and it seems to me like the network is crumbling.

**Comment by btcontract** [comment]
> Since there's a direct channel/connection to the node, it is a bit ridiculous that it cannot find a route. 

Yes, this is strange. BLW invoice does contain routing hints but CL does not need those for a direct channel. This looks like CL issue, probably should be moved there.

**Comment by charltonh** [comment]
I agree CL is dumb and cannot find routes to local channels, but in this case I've dissected the payment info and there is nothing to indicate which channel it goes to.  Also, I believe this is related to the eClair wallet not propagating route information for BLW.
It is very discouraging that BLW doesn't work with CL and no one seems to care.

**Comment by btcontract** [comment]
Not that it matters in this case but BLW invoices always contain a routing tag, also BLW has nothing to do with eclair, it's not a fork of eclair and shares almost no code base with eclair.

**Comment by Sword-Smith** [comment]
I have the same problem. Through the c-lightning CLI the BLW channel (on my phone) is listed as having another `peer_id` than the ones that are in the invoices generated by the BLW.

**Comment by Sword-Smith** [comment]
Please see the following problem description from #c-lightning on IRC Freenode

> 15:47 < sword_smith> I have a weird situation. I have made a channel from the BLW software on my Android phone to my lightning node. I can send from BLW to my c-lightning node but not the other way. The problem                      I get is "Could not find a route". The BLW software generates invoices like this:
> 15:47 < sword_smith> lnbc160n1p0vd9jvdqqcqzgapp5t4qvsg7yhnwmeuajnsyg30vxc86mtvryn3lve3sc4va2v7c3r5qqxqyz5vprzjqf720ngdm34xrj4u76l0n8ulwuwuvce5cef6jr2sjcx3dsl7vu4r6zfw2qqqw8gqqqqqqqfvqqqqqqcqpcpkleuel985jvetfguqgav8uhdks4842ekd3xppart3v6uur425yx42js9447fk0257z0ltx6lc7a43lmgvs662sfew3u550z6veh77spcgefy2
> 15:50 < sword_smith> Parsing this invoice tells me that the payee pub key is 03259c7a586153014e4336d556c7579d69a189eeccbfac5e472010eee3d6dadc5c.
> 15:50 < sword_smith> But I have no channel with this pub key 03259...
> 15:52 < sword_smith> Calling `listfunds` (in c-lightning) the BLW channel looks like it is made to "02216c4b12b5fdf0635eabea6c0867f77e4ecdfe2938e140c1a3514c38cfaa7698".
> 15:52 < sword_smith> Which I cannot find in the BLW-generated invoice.
> 15:54 < darosior> And what tells BLW about its own node id ?..
> 15:57 < sword_smith> Yes, I tried looking for that in the BLW program but I couldn't find it.
> 15:57 < sword_smith> This is the funding txid: f24c9b30f1e1a7d60dcd7c45a895b84f8ff879d9ca0ffc4b78d2ea386d763d45
> 16:02 < sword_smith> c-lightning node ID is: 027ca7cd0ddc6a61cabcf6bef99f9f771dc66334c653a90d50960d16c3fe672a3d

**Comment by Sword-Smith** [comment]
BLW version downloaded from Google Play Store: `0.4.5-151`, c-lightning version: `v0.8.2.1`.

**Comment by btcontract** [comment]
BLW generates decoy node ids in its invoices, could this be the reason? I'd say it's unlikely though since CL should still utilize an embedded routing hint and recognize it as direct channel.

**Comment by Sword-Smith** [comment]
My c-lightning node is connected directly to the running instance of BLW that I have on my Android. I installed BLW on my phone to test my c-lightning node, so all transactions between these two nodes will be done directly, without any nodes in between them.

**Comment by charltonh** [comment]
I just tried closing and reopening my channel from BLW to CL.
The channel opened, then immediately involuntarily closed saying something about "Multiple channel routing unsupported".  This is despite the fact that my CL was built with --enable-experimental-features.  Lost a lot of satoshis in fees.
I'd say it's safe to say BLW is incompatible with CL.

**Comment by charltonh** [comment]
> BLW generates decoy node ids in its invoices, could this be the reason?

decoy node IDs?!   yeah that could be a problem.
